lapce-proxy/src/plugin/psp.rs

```diff
@@ -308,12 +310,15 @@ impl PluginServerRpcHandler {
             check,
             ResponseHandler::Chan(tx),
         );
-        rx.recv().unwrap_or_else(|_| {
-            Err(RpcError {
-                code: 0,
-                message: "io error".to_string(),
+        // TODO: add some kind of logging/visual cue for this timeout issues?
+        // TODO: customizable timeout durations
+        rx.recv_timeout(Duration::from_secs(10))
```

Comment:
It does make the code unreasonable verbose. So perhaps having a timeout here is enough? Perhaps _every_ `recv` needs to be `recv_timeout`? @dzhou121